nilearn/tests/test_signal.py

```diff
@@ -274,6 +287,40 @@ def test_clean_detrending():
     assert_false(abs(x_undetrended - signals).max() < 0.06)
 
 
+def test_clean_t_r():
+    """Different TRs must produce different results after filtering"""
+
+    # n_features  Must be higher than 500
+    for n_samples, n_features in zip(( 34,  42, 100),
+                                     (501, 647, 786)):
+        x_orig = generate_signals_plus_trends(n_features=n_features, n_samples=n_samples)
+
+        random_tr_list1 = np.round(np.random.rand(10)*10, decimals=2)
+        random_tr_list2 = np.round(np.random.rand(10)*10, decimals=2)
```

Comment:
Here you have an unseeded random-number generator. This will create instabilities in the tests that may lead to failures that are hard to reproduce.

The way that we use to avoid this problem is that we create a local random number generator, using "rng = np.random.RandomState(0)". "rng" is then a random number generator from which we can drawn samples using "rng.rand".

This could help stabilize the tests.

Reply:
True. But then we are reducing the test parameter space - sampling only the same few fixed values. I'm just trying to catch all edge cases now than later. cc @bthirion.